modules/apps/29-fee/keeper/escrow.go

```diff
@@ -0,0 +1,139 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"github.com/cosmos/ibc-go/v3/modules/apps/29-fee/types"
+)
+
+// EscrowPacketFee sends the packet fee to the 29-fee module account to hold in escrow
+func (k Keeper) EscrowPacketFee(ctx sdk.Context, identifiedFee *types.IdentifiedPacketFee) error {
+	if !k.IsFeeEnabled(ctx, identifiedFee.PacketId.PortId, identifiedFee.PacketId.ChannelId) {
+		// users may not escrow fees on this channel. Must send packets without a fee message
+		return sdkerrors.Wrap(types.ErrFeeNotEnabled, "cannot escrow fee for packet")
+	}
+	// check if the refund account exists
+	refundAcc, err := sdk.AccAddressFromBech32(identifiedFee.RefundAddress)
+	if err != nil {
+		return err
+	}
+
+	hasRefundAcc := k.authKeeper.GetAccount(ctx, refundAcc)
+	if hasRefundAcc == nil {
+		return sdkerrors.Wrap(types.ErrRefundAccNotFound, fmt.Sprintf("Account with address: %s not found", refundAcc))
+	}
+
+	coins := identifiedFee.Fee.ReceiveFee
+	coins = coins.Add(identifiedFee.Fee.AckFee...)
+	coins = coins.Add(identifiedFee.Fee.TimeoutFee...)
+
+	if err := k.bankKeeper.SendCoinsFromAccountToModule(
+		ctx, refundAcc, types.ModuleName, coins,
+	); err != nil {
+		return err
+	}
+
+	// Store fee in state for reference later
+	k.SetFeeInEscrow(ctx, identifiedFee)
```

Comment:
It looks like this will override any fee that are already registered in the store, if `EscrowPacketFee` is called multiple times with the same `PacketId`.